chart/templates/deployment.yaml

```diff
@@ -8,11 +8,11 @@ spec:
   revisionHistoryLimit: 2
   selector:
     matchLabels:
-      app: hcloud-cloud-controller-manager
+      {{- include "hcloud-cloud-controller-manager.selectorLabels" . | nindent 6 }}
```

Comment:
The selector is [immutable](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector):

> Also note that .spec.selector is immutable after creation of the Deployment in apps/v1.

I think we need to keep `app` in there, or we would need to declare a breaking change, as users would need to re-create the deployment.

Reply:
We can pretend that Helm charts aren't stable versioned artifacts, right? Since we haven't documented them yet and our Helm repo has a big scary warning that it's not ready yet (and it's not!).

If so, then I think the cleanest workaround to this is to make the selectorLabels configurable in values.yaml. The static (legacy-ish) manifests we generate can continue to use the old under-specified `app` label, whilst usage of the Helm chart directly defaults to the "modern" (using the standard labels + distinguishing multiple releases) labels.

WDYT?

Reply:
Sounds good :+1:

Reply:
Done. Re-ran `generate-deployment-yamls.sh` and as you can see it reshuffled things a bit (and sneakily introduced the metrics container port) but the selector labels has been left alone.